lib/rubygems/remote_fetcher.rb

```diff
@@ -78,7 +77,7 @@ def self.fetcher
   # +headers+: A set of additional HTTP headers to be sent to the server when
   #            fetching the gem.
 
-  def initialize(proxy=nil, dns=nil, headers={})
+  def initialize(proxy=nil, headers={})
```

Comment:
@simi Do you know a good way to gracefully deprecate the removal of one parameter? :thinking: I don't think this is public API but still.

Reply:
My initial idea is to handle arguments manually.

```ruby
def deprecated_initialize(*arguments)
  if arguments.length > 3
    raise ArgumentError, "(wrong number of arguments (given #{arguments.length}, expected 0..3))"
  end

  if arguments.length == 3
    warn "dns is not used anymore deprecation warning"
    proxy = arguments[0]
    headers = arguments[2]
  else
    proxy = arguments[0]
    headers = arguments[1]
  end

  headers ||= {} # assign default value

  p "proxy: #{proxy}"
  p "headers: #{headers}"
end
```

```ruby
deprecated_initialize("proxy", "dns", "headers") # old API
#=> "dns is not used anymore deprecation warning"
#=> "proxy: proxy"
#=> "headers: headers"

deprecated_initialize("proxy", "headers") # new API
#=> "proxy: proxy"
#=> "headers: headers"

deprecated_initialize("proxy") # new API + default value
#=> "proxy: proxy"
#=> "headers: {}"


```

It is not 100% (it works differently with explicitly passed `nil` as headers). :thinking: But is that really worth the complexity in here?

Reply:
I was just curious how you'd handle it, that seems like it would be the way to go yeah, thanks!

I think this is fine to just remove or otherwise just leave the `dns` parameter in there for now since it's actually unrelated from this PR, I only want to stop requiring `resolv`.

Reply:
:muscle: